src/NLog/Targets/EventLogTarget.cs

```diff
@@ -349,7 +349,9 @@ internal string GetFixedSource()
         /// <returns></returns>
         private EventLog GetEventLog(LogEventInfo logEvent)
         {
-            return eventLogInstance ?? (eventLogInstance = new EventLog(this.Log, this.MachineName, this.Source.Render(logEvent)));
+            if (eventLogInstance?.Source == this.Source.Render(logEvent))
```

Comment:
Think it fixes the issue (hard to read on mobile) but Its still double checking null (`?.` and `==`), which is a bit sloppy